lib/components/primitive-components/Trace/Trace.ts

```diff
@@ -658,18 +659,33 @@ export class Trace
       .list()
       .find((label) => label.source_net_id === toPort.source_port_id)
 
-    if (
-      (existingFromNetLabel &&
-        existingFromNetLabel.text !== this.props.schDisplayLabel) ||
-      (existingToNetLabel &&
-        existingToNetLabel?.text !== this.props.schDisplayLabel)
-    ) {
-      throw new Error(
-        `Cannot create net label for port ${existingFromNetLabel ? fromPortName : toPortName} because it already has a net label with text "${existingFromNetLabel ? existingFromNetLabel.text : existingToNetLabel?.text}".`,
-      )
+    if (this.props.schDisplayLabel) {
+      if (
+        (existingFromNetLabel &&
+          existingFromNetLabel.text !== this.props.schDisplayLabel) ||
+        (existingToNetLabel &&
+          existingToNetLabel?.text !== this.props.schDisplayLabel)
+      ) {
+        throw new Error(
+          `Cannot create net label for port ${existingFromNetLabel ? fromPortName : toPortName} because it already has a net label with text "${existingFromNetLabel ? existingFromNetLabel.text : existingToNetLabel?.text}".`,
+        )
+      }
     }
+
+    const [firstPort, secondPort] = connectedPorts.map(({ port }) => port)
+    const isFirstPortSchematicBox =
+      firstPort.parent?.config.shouldRenderAsSchematicBox
+    const isSecondPortSchematicBox =
+      secondPort.parent?.config.shouldRenderAsSchematicBox
+    const firstPortName = isFirstPortSchematicBox
+      ? firstPort?.props.name
+      : `${secondPort?.parent?.props.name}_${secondPort?.props.name}`
+    const secondPortName = isSecondPortSchematicBox
+      ? secondPort?.props.name
+      : `${firstPort?.parent?.props.name}_${firstPort?.props.name}`
+
```

Comment:
The names should always be identical (its just a schematic convention for net labels), if both parents are schematic boxes we could do a joint name (“U1_SDA_U2_SCL”)

i know thats pretty gross/long but i cant think of a better solution for two boxes at the moment.

Reply:
@seveibar  The code is currently functioning only for passive-to-chip connections, as indicated by the issue. Should I modify it to handle all trace types, including passive-to-passive and chip-to-chip connections as well?

Reply:
@seveibar

Reply:
no passive to chip is fine to start!

Reply:
but the labels should be the same on each side!